components/view_image.jsx

```diff
@@ -318,18 +345,6 @@ export default class ViewImageModal extends React.Component {
     }
 }
 
-ViewImageModal.defaultProps = {
-    show: false,
-    fileInfos: [],
-    startIndex: 0
-};
-ViewImageModal.propTypes = {
-    show: PropTypes.bool.isRequired,
-    onModalDismissed: PropTypes.func.isRequired,
-    fileInfos: PropTypes.arrayOf(PropTypes.object).isRequired,
-    startIndex: PropTypes.number
-};
-
 function LoadingImagePreview({progress, loading}) {
```

Comment:
I suggest the following to be on a separate file as a separate pure component with corresponding component tests:
- LoadingImagePreview to loading_image_preview.jsx
- ImagePreview to image_preview.jsx

Reply:
@saturninoabril Separating component files is a good idea. They don't have a state, so can I leave it in a stateless-functional style, not a pure component?

Reply:
Yes, it'd be fine as stateless functional component.